mos_tests/rabbitmq_oslo/test_restarts.py

```diff
@@ -0,0 +1,156 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def vars_config(remote):
+    """Prepare variables and different paths
+    :param remote: SSH connection point to controller.
+    """
+    config_vars = {
+        'repo': 'https://github.com/dmitrymex/oslo.messaging-check-tool.git',
+        'pkg': 'oslo.messaging-check-tool_1.0-1~u14.04+mos1_all.deb',
+        'root_path': '/root/',
+        'repo_path': '/root/oslo.messaging-check-tool/',
+        'nova_user': 'nova'}
+    config_vars['conf_file_path'] = '{}oslo_msg_check.conf'.format(
+        config_vars['repo_path'])
+    # get password of nova user (the same on all controllers)
+    cmd = "grep '^rabbit_password' /etc/nova/nova.conf | awk '{print $3}'"
+    config_vars['nova_pass'] = remote.check_call(cmd)['stdout'][0].strip()
+    return config_vars
+
+
+def install_oslomessagingchecktool(remote, **kwargs):
+    """Install tool on controller
+    https://github.com/dmitrymex/oslo.messaging-check-tool
+    """
+    cmd = ("apt-get update && "
+           "apt-get install git python-pip -y && "
+           "cd {root_path} && "
+           "git clone {repo} && "
+           "cd {repo_path} && "
+           "pip install -r requirements.txt -r test-requirements.txt && "
+           "dpkg -i {pkg} || "
+           "apt-get -f install -y".format(root_path=kwargs['root_path'],
+                                          repo=kwargs['repo'],
+                                          repo_path=kwargs['repo_path'],
+                                          pkg=kwargs['pkg']))
+    logger.debug('Install "oslo.messaging-check-tool" on controller')
+    return remote.check_call(cmd)
+
+
+def configure_oslomessagingchecktool(remote, ctrl_ips, **kwargs):
+    """Write configuration file on controller"""
+    # Create config file for "oslo.messaging-check-tool"
+    configs = (
+        "[DEFAULT]\n"
+        "debug=true\n"
+        "[oslo_messaging_rabbit]\n"
+        "rabbit_hosts = {rabbit_hosts}\n"
+        "rabbit_userid = {nova_user}\n"
+        "rabbit_password = {nova_pass}\n".format(
+            rabbit_hosts=":5673, ".join(ctrl_ips),
+            nova_user=kwargs['nova_user'],
+            nova_pass=kwargs['nova_pass']))
+    # Write config to file on controller
+    logger.debug('Write "oslo.messaging-check-tool" config file to controller')
+    with remote.open(kwargs['conf_file_path'], 'w') as f:
+        f.write(configs)
+
+
+def restart_rabbitmq_serv(env, remote=None, restart_all=False):
+    """Restart rabbitmq-server service on one or all controllers
+    :param env: Environment
+    :param remote: SSH connection point to controller.
+        Leave empty if you want to restart service on all controllers.
+    :param restart_all: Set True to restart on service on all controllers.
+    :return: None
+    """
+    restart_cmd = 'service rabbitmq-server restart'
+    # if certain controller not set --> restart rabbitmq on all controllers
+    if remote is None:
+        restart_all = True
```

Comment:
You change variable and then check it again and restarts all services. Why you don't restart services on all controllers right here? What goal of `restart_all` variable? In this code `restart_all` is just equal `remote is None`